engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailProcessor.java

```diff
@@ -0,0 +1,181 @@
+import com.redhat.cloud.notifications.processors.email.connector.dto.EmailNotification;
+import com.redhat.cloud.notifications.processors.email.connector.dto.RecipientSettings;
+import com.redhat.cloud.notifications.transformers.BaseTransformer;
+import io.quarkus.logging.Log;
+import io.vertx.core.json.JsonObject;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class EmailProcessor extends SystemEndpointTypeProcessor {
+    @Inject
+    BaseTransformer baseTransformer;
+
+    @Inject
+    ConnectorSender connectorSender;
+
+    @Inject
+    EndpointRepository endpointRepository;
+
+    @Inject
+    EmailAggregationRepository emailAggregationRepository;
+
+    @Inject
+    NotificationHistoryRepository notificationHistoryRepository;
+
+    @Inject
+    TemplateRepository templateRepository;
+
+    /**
+     * The non-instant subscription types that we support.
+     */
+    private static final List<EmailSubscriptionType> NON_INSTANT_SUBSCRIPTION_TYPES =
+        Arrays
+            .stream(EmailSubscriptionType.values())
+            .filter(emailSubscriptionType -> emailSubscriptionType != EmailSubscriptionType.INSTANT)
+            .collect(Collectors.toList());
+
+
+    @Override
+    public void process(final Event event, final List<Endpoint> endpoints) {
+        if (endpoints == null || endpoints.isEmpty()) {
+            return;
+        }
+
+        final EventType eventType = event.getEventType();
+        final String bundleName = eventType.getApplication().getBundle().getName();
+        final String applicationName = eventType.getApplication().getName();
+        final String accountId = event.getAccountId();
+        final String orgId = event.getOrgId();
+
+        // Check if the aggregation should be persisted or not.
+        if (this.templateRepository.isEmailAggregationSupported(bundleName, applicationName, NON_INSTANT_SUBSCRIPTION_TYPES)) {
+            final EmailAggregation aggregation = new EmailAggregation();
+            aggregation.setApplicationName(applicationName);
+            aggregation.setBundleName(bundleName);
+            aggregation.setOrgId(orgId);
+
+            // Transform the event to
+            final JsonObject transformedEvent = this.baseTransformer.toJsonObject(event);
+            aggregation.setPayload(transformedEvent);
+
+            try {
+                this.emailAggregationRepository.addEmailAggregation(aggregation);
+            } catch (Exception e) {
+                // ConstraintViolationException may be thrown here and it must not interrupt the email that is being sent.
+                Log.warn("Email aggregation persisting failed", e);
+            }
+        }
```

Comment:
Most of this code is duplicated from `EmailSubscriptionTypeProcessor`. Could we expose the old code with a `public` method and use it from here instead instead of duplicating it?

Reply:
All right. I left the other one untouched because even though I started refactoring everything, I thought that it might have been easier to simply drop that processor after we switch to the connector. But I've followed your suggestion now.